components/director/internal/domain/labeldef/service.go

```diff
@@ -155,3 +165,53 @@ func (s *service) validateExistingLabelsAgainstSchema(ctx context.Context, schem
 	}
 	return nil
 }
+
+func (s *service) validateAutomaticScenarioAssignmentAgainstSchema(ctx context.Context, schema interface{}, tenantID, key string) error {
+	if key != model.ScenariosKey {
+		return nil
+	}
+
+	validator, err := jsonschema.NewValidatorFromRawSchema(schema)
+	if err != nil {
+		return errors.Wrap(err, "while creating validator for a new schema")
+	}
+	inUse, err := s.fetchScenariosFromAssignments(ctx, tenantID)
+	if err != nil {
+		return err
+	}
+	for _, used := range inUse {
+		res, err := validator.ValidateRaw([]interface{}{used})
+		if err != nil {
+			return errors.Wrapf(err, "while validating scenario assignment [scenario=%s] with a new schema", used)
+		}
+		if res.Error != nil {
+			return errors.Wrapf(res.Error, "Scenario Assignment [scenario=%s] is not valid against a new schema", used)
+		}
+	}
+	return nil
+}
+
+func (s *service) fetchScenariosFromAssignments(ctx context.Context, tenantID string) ([]string, error) {
+	m := make(map[string]struct{})
```

Comment:
To discuss, but maybe it is worth implementing a service method that will return you what you need directly from the database instead of doing this in the source code?